File: app/src/main/java/com/example/pie/android/model/TodoItem.java
package com.example.pie.android.model;

import com.google.gson.annotations.Expose;

public class TodoItem {

    /**
     * The Server ID
     */
    @Expose(serialize = false)
    private int id;

    /**
     * The Cient ID - Should never be sent to the server.
     */
    private long _id;

    @Expose
    private String task;

    @Expose
    private boolean done;

    public long get_id() {
        return _id;
    }

    public void set_id(long _id) {
        this._id = _id;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getTask() {
        return task;
    }

    public void setTask(String task) {
        this.task = task;
    }

    public boolean isDone() {
        return done;
    }

    public void setDone(boolean done) {
        this.done = done;
    }

    public void complete(){
        this.done = true;
    }
}

File: app/src/main/java/com/example/pie/android/model/TodoItem.kt
package com.example.pie.android.model

import android.os.Parcel
import com.example.pie.android.util.KParcelable
import com.example.pie.android.util.parcelableCreator
import com.google.gson.annotations.Expose

data class TodoItem (

    /**
     * The Server ID
     */
    @Expose(serialize = false)
    var id: Int = 0,

    /**
     * The Client ID - Should never be sent to the server.
     */
    var _id: Long = 0,

    @Expose
    var task: String? = null,

    @Expose
    var done: Boolean = false
): KParcelable {

    constructor(parcel: Parcel): this(
            id = parcel.readInt(),
            _id = parcel.readLong(),
            task = parcel.readString(),
            done = parcel.readByte() != 0.toByte()
    )

    override fun writeToParcel(parcel: Parcel, flags: Int) {
        with(parcel) {
            writeInt(id)
            writeLong(_id)
            writeString(task)
            writeByte(if (done) 1 else 0)
        }
    }

    fun complete() {
        this.done = true
    }

    companion object {
        @JvmField
        val CREATOR = parcelableCreator(::TodoItem)
    }
}